components/layout_2020/style_ext.rs

```diff
@@ -58,6 +60,23 @@ impl ComputedValuesExt for ComputedValues {
         (writing_mode, direction)
     }
 
+    fn writing_mode_is_horizontal(&self) -> bool {
+        match self.get_inherited_box().writing_mode {
+            WritingMode::HorizontalTb => true,
+            WritingMode::VerticalLr | WritingMode::VerticalRl => false,
+        }
+    }
+
+    fn inline_size_is_auto(&self) -> bool {
+        let position = self.get_position();
+        let size = if self.writing_mode_is_horizontal() {
+            position.width
+        } else {
+            position.height
+        };
+        matches!(size, Size::Auto)
+    }
```

Comment:
Why not just do `self.box_size().inline == LengthOrAuto::Auto`? At the very least, I don't think the `matches!` is necessary here.

Reply:
It seems like more work to also compute the block_size only to drop it. But anyway since https://doc.servo.org/style/properties/style_structs/struct.Position.html#structfield.inline_size & co exist in memory we should remove all of these methods and access those fields instead. But let’s keep that change for a future PR